hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SingleTransactionBlockItemsBuilderImpl.java

```diff
@@ -0,0 +1,1189 @@
+import com.hedera.node.app.state.SingleTransactionRecord.TransactionOutputs;
+import com.hedera.pbj.runtime.OneOf;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.crypto.DigestType;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Block items equivalent of {@link SingleTransactionRecordBuilderImpl}.
+ */
+@SuppressWarnings({"unused", "UnusedReturnValue"})
+public class SingleTransactionBlockItemsBuilderImpl
+        implements SingleTransactionBuilder,
+                ConsensusCreateTopicRecordBuilder,
+                ConsensusSubmitMessageRecordBuilder,
+                CreateFileRecordBuilder,
+                CryptoCreateRecordBuilder,
+                CryptoTransferRecordBuilder,
+                ChildRecordBuilder,
+                PrngRecordBuilder,
+                ScheduleRecordBuilder,
+                TokenMintRecordBuilder,
+                TokenBurnRecordBuilder,
+                TokenCreateRecordBuilder,
+                ContractCreateRecordBuilder,
+                ContractCallRecordBuilder,
+                ContractUpdateRecordBuilder,
+                EthereumTransactionRecordBuilder,
+                CryptoDeleteRecordBuilder,
+                TokenUpdateRecordBuilder,
+                NodeStakeUpdateRecordBuilder,
+                FeeBuilder,
+                ContractDeleteRecordBuilder,
+                GenesisAccountRecordBuilder,
+                ContractOperationRecordBuilder,
+                TokenAccountWipeRecordBuilder,
+                CryptoUpdateRecordBuilder,
+                NodeCreateRecordBuilder {
+    private static final Comparator<TokenAssociation> TOKEN_ASSOCIATION_COMPARATOR =
+            Comparator.<TokenAssociation>comparingLong(a -> a.tokenId().tokenNum())
+                    .thenComparingLong(a -> a.accountIdOrThrow().accountNum());
+    // base transaction data
+    private Transaction transaction;
+    private Bytes transactionBytes = Bytes.EMPTY;
+    // fields needed for TransactionRecord
+    // Mutable because the provisional consensus timestamp assigned on dispatch could
+    // change when removable records appear "between" this record and the parent record
+    private Instant consensusNow;
+    private Instant parentConsensus;
+    private TransactionID transactionID;
+    private List<TokenTransferList> tokenTransferLists = new LinkedList<>();
+    private List<AssessedCustomFee> assessedCustomFees = new LinkedList<>();
+    private List<TokenAssociation> automaticTokenAssociations = new LinkedList<>();
+
```

Comment:
Is this same as `SingleTransactionRecordBuilderImpl` ?

Reply:
I copy-pasted this initially. I've taken a lot more out of this class so they aren't the same now